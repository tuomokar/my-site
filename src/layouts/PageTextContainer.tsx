import * as React from 'react'
import styled from 'styled-components'

interface TextPageContainerProps {
  children: React.ReactNode
}

const Container = styled.div`
  padding-top: 50px;
`

const PageTextContainer: React.FC<TextPageContainerProps> = ({ children }) => (
  <Container>{children}</Container>
)

export default PageTextContainer
