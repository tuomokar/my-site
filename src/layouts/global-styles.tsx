import * as React from 'react'
import { createGlobalStyle } from 'styled-components'

const GlobalStyle = createGlobalStyle`
  body {
    margin: 0;
  }
`

const GlobalStyles: React.FC = () => <GlobalStyle />

export default GlobalStyles
