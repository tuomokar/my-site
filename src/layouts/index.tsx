import PageContainer from './page-container'

/*
 * Is used by gatsby-plugin-layout to wrap every page within this component.
 */
export default PageContainer
