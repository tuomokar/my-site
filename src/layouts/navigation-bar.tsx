import * as React from 'react'
import Navbar from 'react-bootstrap/Navbar'
import Nav from 'react-bootstrap/Nav'

import Link from 'gatsby-link'

const NavigationBar: React.FC = () => (
  <Navbar bg="dark" variant="dark">
    <Link to="/">
      <Navbar.Brand>Tuomo Oila</Navbar.Brand>
    </Link>
    <Nav>
      <Link to="/biography">
        <Nav.Link as="span">Bio</Nav.Link>
      </Link>
      <Link to="/work">
        <Nav.Link as="span">Work</Nav.Link>
      </Link>
      <Link to="/contact">
        <Nav.Link as="span">Contact</Nav.Link>
      </Link>
    </Nav>
  </Navbar>
)

export default NavigationBar
