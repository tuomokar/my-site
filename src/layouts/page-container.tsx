import * as React from 'react'
import styled from 'styled-components'

import GlobalStyles from './global-styles'

interface ContainerProps {
  children: React.ReactNode
}

const PageContentContainer = styled.div`
  height: 100vh;
`

const PageContainer: React.FC<ContainerProps> = ({ children }) => (
  <React.Fragment>
    <GlobalStyles />
    <PageContentContainer>{children}</PageContentContainer>
  </React.Fragment>
)

export default PageContainer
