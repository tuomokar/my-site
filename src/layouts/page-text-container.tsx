import * as React from 'react'
import styled from 'styled-components'

interface TextPageContainerProps {
  children: React.ReactNode
}

const Container = styled.div`
  padding: 50px 10px 10px;
  max-width: 900px;
  margin: 0 auto;
`

const PageTextContainer: React.FC<TextPageContainerProps> = ({ children }) => (
  <Container>{children}</Container>
)

export default PageTextContainer
