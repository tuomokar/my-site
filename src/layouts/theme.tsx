import * as React from 'react'
import { ThemeProvider } from 'styled-components'

const theme = {
  colors: {
    primaryBlue: '#1A4782',
    primaryGreen: '#68BA6E',
  },
}

interface ThemeProps {
  children: React.ReactNode
}

const Theme: React.FC<ThemeProps> = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
)

export default Theme
