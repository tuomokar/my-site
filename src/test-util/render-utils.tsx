import * as React from 'react'
import { render } from '@testing-library/react'

import Theme from '../layouts/theme'

export const renderWithTheme = (children: React.ReactNode) =>
  render(<Theme>{children}</Theme>)
