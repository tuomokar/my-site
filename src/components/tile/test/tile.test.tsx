import * as React from 'react'

import Tile from '../tile'
import { renderWithTheme } from '../../../test-util/render-utils'
import { theme } from '../../../layouts/theme'
// Side effect import
// TODO: not having the import here (and only in jest.setup.js) breaks typescript (the matchers aren't added to jest)
import 'jest-styled-components'

const initialContentText = 'initialContentTestText'
const extraContentText = 'extraContentTestText'

const mockProps: React.ComponentProps<typeof Tile> = {
  initialContent: initialContentText,
  extraContent: extraContentText,
  color: 'primaryBlue',
}

const renderTile = (props = mockProps) => renderWithTheme(<Tile {...props} />)

describe('tile', () => {
  it('renders the initial content given', () => {
    const { getByText } = renderTile()

    expect(getByText(initialContentText)).toBeInTheDocument()
  })

  it('renders with the given color as the background color', () => {
    const { container } = renderTile()

    expect(container.firstChild).toHaveStyleRule(
      `background-color`,
      theme.colors[mockProps.color]
    )
  })
})
