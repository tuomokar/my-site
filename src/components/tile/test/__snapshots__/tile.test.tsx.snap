// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tile renders correctly 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: white;
  height: 100%;
  font-size: 30px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 {
  position: absolute;
  width: 50%;
  height: 50%;
  background-color: #1A4782;
}

<div>
  <div
    class="c0"
    color="primaryBlue"
  >
    <div
      class="c1"
    >
      <div>
        initialContentTestText
      </div>
      extraContentTestText
    </div>
  </div>
</div>
`;
